src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,140 @@
+     *
+     * @param index The index of the task to be marked as done
+     */
+    public void markDone(int index) throws DukeException {
+        try {
+            Task task = this.taskList.get(index - 1);
+            task.setDone();
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException("☹ OOPS!!! There is no task with this index number.");
+        }
+    }
+
+    /**
+     * Marks the task with the specified index as not done.
+     *
+     * @param index The index of the task to be marked as not done
+     */
+    public void markNotDone(int index) throws DukeException {
+        try {
+            Task task = this.taskList.get(index - 1);
+            task.setNotDone();
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException("☹ OOPS!!! There is no task with this index number.");
+        }
+    }
+    /**
+     * Removes the task with the specified index from the task list.
+     *
+     * @param number The index of the task to be removed.
+     */
+    public void delete(int number) throws DukeException {
+        try {
+            this.index = this.taskList.size();
+            this.taskList.remove(number - 1);
+            this.index--;
+
+        } catch (IndexOutOfBoundsException | NullPointerException e) {
+            throw new DukeException("☹ OOPS!!! There is no task with this index number.");
+        }
+    }
+
+    public ArrayList<Task> getList() {
+        return this.taskList;
+    }
+
+    public int getIndex() {
+        return this.index;
+    }
+
+    public Task getTask(int index) throws DukeException {
+        try {
+            return this.taskList.get(index - 1);
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException("☹ OOPS!!! There is no task with this index number.");
+        }
+    }
+
+    /**
+     * Prints the tasks in the list.
+     */
+    public void printTaskList() {
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < this.getSize(); i++) {
+            System.out.println((i + 1) + ". " + this.taskList.get(i).displayableForm());
+        }
+    }
+
+    /**
+     * Finds tasks in the list that contain the keyword.
+     * @param keyword The keyword input by the user.
+     * @return An array list containing the tasks that contain the key word.
+     * @throws DukeException
+     */
+    public ArrayList<Task> findTasks(String keyword) throws DukeException {
+        try {
+            ArrayList<Task> tempList = new ArrayList<>(this.getSize());
+            for (int i = 0; i < this.getSize(); i++) {
+                if (this.taskList.get(i).getName().contains(keyword)) {
+                    tempList.add(this.taskList.get(i));
+                }
+            }
+            return tempList;
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException("☹ OOPS!!! There are no tasks in the list.");
+        }
+    }
+
+    public int getSize() {
```

Comment:
Given that you've meticulously provided JavaDoc comments for all classes and methods elsewhere, is it intentional that there are missing JavaDoc comments in this particular method?